packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol

```diff
@@ -0,0 +1,1194 @@
+
+    /// @notice Thrown when a config load fails.
+    error OPContractsManagerV2_ConfigLoadFailed(string _name);
+
+    /// @notice Container of blueprint and implementation contract addresses.
+    IOPContractsManagerContainer public immutable contractsContainer;
+
+    /// @notice Address of the Standard Validator for this OPCM release.
+    IOPContractsManagerStandardValidator public immutable standardValidator;
+
+    /// @notice The version of the OPCM contract.
+    string public constant version = "6.0.0";
+
+    /// @param _contractsContainer The container of blueprint and implementation contract addresses.
+    /// @param _standardValidator The standard validator for this OPCM release.
+    constructor(
+        IOPContractsManagerContainer _contractsContainer,
+        IOPContractsManagerStandardValidator _standardValidator
+    ) {
+        contractsContainer = _contractsContainer;
+        standardValidator = _standardValidator;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    //                   PUBLIC CHAIN MANAGEMENT FUNCTIONS                   //
+    ///////////////////////////////////////////////////////////////////////////
+
+    /// @notice Upgrades the Superchain contracts. Currently this is limited to the
+    ///         SuperchainConfig contract, but may eventually expand to include other
+    ///         Superchain-wide contracts.
+    /// @param _inp The input for the Superchain upgrade.
+    function upgradeSuperchain(SuperchainUpgradeInput memory _inp) external returns (SuperchainContracts memory) {
+        // NOTE: Since this function is very minimal and only upgrades the SuperchainConfig
+        // contract, not bothering to fully follow the pattern of the normal chain upgrade flow.
+        // If we expand the scope of this function to add other Superchain-wide contracts, we'll
+        // probably want to start following a similar pattern to the chain upgrade flow.
+
+        // Upgrade the SuperchainConfig if it has changed.
+        _upgrade(
+            IProxyAdmin(_inp.superchainConfig.proxyAdmin()),
+            address(_inp.superchainConfig),
+            implementations().superchainConfigImpl,
+            abi.encodeCall(ISuperchainConfig.initialize, (_inp.superchainConfig.guardian()))
+        );
+
+        // Return the Superchain contracts.
+        return SuperchainContracts({ superchainConfig: _inp.superchainConfig });
+    }
+
+    /// @notice Deploys a new chain from full config.
+    /// @param _cfg The full chain deployment configuration.
+    /// @return The chain contracts.
+    function deploy(FullConfig memory _cfg) external returns (ChainContracts memory) {
+        // Deploy is the ONLY place where we allow the "ALL" permission for proxy deployment.
+        ExtraInstruction[] memory instructions = new ExtraInstruction[](1);
+        instructions[0] = ExtraInstruction({ key: "PermittedProxyDeployment", data: bytes("ALL") });
```

Comment:
I think that this has become slightly confusing now that fd308b4ec0428f1d9f5718ba9cdde5e22677622a has added an ExtraInstruction field to FullConfig, ie. there is :
1. `_cfg.extraInstructions` provided as an arg to `deploy()` which is passed into `_execute()`, and 
2. `instructions` created here and passed into `_loadChainContracts()`

WDYT if `_loadChainContracts()` accepted a boolean instead, and then use that boolean to create the `PermittedProxyDeployment: ALL` instruction inline?